sdk/src/lib.rs

```diff
@@ -62,18 +62,138 @@ impl<'a> TpProcessRequest<'a> {
     }
 }
 
-pub struct TransactionContext {}
+pub trait TransactionContext {
+    #[deprecated(
+        since = "0.2.0",
+        note = "please use `get_state_entry` or `get_state_entries` instead"
+    )]
+    /// get_state queries the validator state for data at each of the
+    /// addresses in the given list. The addresses that have been set
+    /// are returned. get_state is deprecated, please use get_state_entry or get_state_entries
```

Comment:
Thoughts on using #deprecated? https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-190-2016-05-26

Reply:
#deprecated is already used right above this comment

Reply:
Hah! Maybe that makes the additional comments redundant. Thoughts?

Reply:
This currently matches what is in the rust sdk but I would be okay with removing the depreciated part of the doc comment if that is desired. This is what it looks like when docs are generated.
![Screen Shot 2019-04-05 at 8 32 10 AM](https://user-images.githubusercontent.com/18215486/55631314-60519a80-577d-11e9-85e4-319220c19e22.png)

Reply:
Ok, sounds good as is.